### DISCUSSION TITLE
Error: connect ECONNREFUSED ::1:61205

### DISCUSSION THREAD

**Discussion by druoid** [pact-foundation/pact-js#1083] [Q&A] [answered]
Not sure how to resolve this and looking for some help.

Currently trying to get the 5 minute guide from https://docs.pact.io/5-minute-getting-started-guide to work.

Cloned the repo and made no changes to the code 

When running npm run test:consumer I get the following error:

`Error: connect ECONNREFUSED ::1:61205
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1494:16) {
  errno: -61,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '::1',
  port: 61205,
  response: undefined
}`

Using a Mac OS Monterey 12.5.1
NPM version 9.6.4
Node version 18.15.0

https://github.com/druoid/pact-5-minute-getting-started-guide

**Comment by mefellows** [comment, marked as answer]
Looks like an IPv6 issue. I've just cloned it and get the same error. Note the address is `::1`. Replacing all references of `localhost` to `127.0.0.1` (and ipv4 address) fixes it. You'll need to do it in both the consumer and provider. 

I'll see about getting the guide updated.

**Reply by druoid** [reply]
Thanks for that.

**Reply by mefellows** [reply]
See also https://github.com/pact-foundation/pact-5-minute-getting-started-guide/compare/chore/ipv6-for-mac-montery?expand=1

**Reply by YOU54F** [reply]
Hi @druoid were you running an m1 or intel machine?

You can find out by typing `arch` into your terminal?

Happy to put a note in at the moment in the comments to help out someone out in the interim, sorry that you got stuck!

**Reply by YOU54F** [reply]
If we bind to `127.0.0.1` it causes a web view to appear in the repl, which stops it appearing in users browsers, it's a pain!

https://replit.com/talk/ask/Can-we-make-the-web-interface-go-away/35726

It's an issue thats been around for a good while

**Comment by mefellows** [comment]
This project doesn't use jasmine, so that will have no impact. See the suggestion above.

**Comment by druoid** [comment]
Hi Yousaf,

I'm running on Apple M1 Pro

Regards

On Thu, Apr 27, 2023 at 11:42 AM Yousaf Nabi ***@***.***>
wrote:

> Hi @druoid <https://github.com/druoid> were you running an m1 or intel
> machine?
>
> You can find out by typing arch into your terminal?
>
> Happy to put a note in at the moment in the comments to help out someone
> out in the interim, sorry that you got stuck!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/pact-foundation/pact-js/discussions/1083#discussioncomment-5736900>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A6WJ3CHCQ2MXYBF3LSZ6VSLXDGXGHANCNFSM6AAAAAAWYQ4DAY>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***
> com>
>


-- 
Andrew Chambers

6 Lerwick Terrace
Melrose
Wellington
6023


Mob: NZ 021 0705321

**Comment by Jesus-David-Vidal-Ramirez** [comment]
Me esta generando este error, cuando me conecto a una base de datos local me puedo conectar bien pero a una remota no y desde table plus puedo entrar de lo mas de bien en ambas
Error: connect ECONNREFUSED ::1:3907
    at __node_internal_captureLargerStackTrace (node:internal/errors:490:5)
    at __node_internal_exceptionWithHostPort (node:internal/errors:668
